docs/DeveloperGuide.md

```diff
@@ -144,6 +144,104 @@ of the system.
+
+![InsertClassDiagram](diagrams/OrGateClassDiagram.png)
+
+
+There are four Boolean commands that are used in the implementation of the logic gates: `TemplateBooleanCommand, SetBooleanCommand, AddBooleanCommand, and CalcBooleanCommand`.
+
+####SetBooleanCommmand
+The `setBooleanCommand` is used to set the value of one of two inputs to the logic gates, which can be either the Boolean value of `0 or 1`.
+
+1. It is instantiated through the `execute()` function in the Parser class.
+2. The `setBooleanCommand` object will then call the `setInput(template logicGate, String gateInput, int value)` method which will take in a template, input of a logic gate as `String` data type, 
+and its Boolean value to be set as its three inputs. The function takes in a template as its first input and will instantiate the BooleanTemplate object.
+3. The `BooleanTemplate` object will then convert the input of the logic gate into the index to be stored in the binary tree of logic gates. This will depend on where the 
+logic gate is stored in the binary tree. This can be achieved by passing in the gate in `String` form to the `convertToIndex("C")` function, where in this case `"C"` is the
+input of the logic gate to be converted.
+4. The `BooleanTemplate` object will then call `insertTree` function in the instantiated `BinaryTree` object, where the `insertTree` function will take in the index of the logic gate
+stored in the `BinaryTree` and the Boolean value to be assigned to the gate as input. The binary tree containing the logic gates is stored in this object.
+5. The `BinaryTree` object will then call on the `setInput` function in the instantiated `OrGate` object with `index` of the logic gate and its Boolean value to be
+assigned as inputs.
+6. The `OrGate` object represents the `or` operation logic gate. 
+It then calls `setBoolean(int value)` function on itself to assign the Boolean value to that selected input of the logic gate.
+
+The above sequence of object interactions through the SetCommand can be represented in the following sequence diagram:
+
+![InsertSequeunceDiagram](diagrams/SetBooleanCommand.png)
+
+####AddBooleanCommmand
+
+The `AddBooleanCommand` is used to combine multiple logic gate templates to produce advanced Boolean logic gate configurations.
+For instance, an `OrGate` can be combined with an `AndGate` to produce a new logic gate where its final output will depend on the
+Boolean values assigned to the `OrGate` and `AndGate`. The combination of both these gates can be represented by the following object diagram:
+
+![InsertObjectDiagram](diagrams/AddOrAndGates.png)
+
+The above combined logic gate configuration can also further be combined with another logic gate template (for instance, an `XorGate`) using the `AddBooleanCommand`
+again to produce an even more advanced logic gate configuration. This combination can be represented by the following
+object diagram:
+
+![InsertObjectDiagram](diagrams/AddAndOrXORGates.png)
```

Comment:
Where is the PUML file for this? It seems like the `TwoInputGate` objects shouldn't know about it each other, rather it's `BooleanTemplate` that knows about all of them.